packages/backend/src/open_payments/errors.ts

```diff
@@ -0,0 +1,65 @@
+import { AppContext } from '../app'
+import { OpenAPIValidatorMiddlewareError } from '@interledger/openapi'
+
+export class OpenPaymentsServerRouteError extends Error {
+  public status: number
+  public details?: Record<string, unknown>
+
+  constructor(
+    status: number,
+    message: string,
+    details?: Record<string, unknown>
+  ) {
+    super(message)
+    this.name = 'OpenPaymentsServerRouteError'
+    this.status = status
+    this.details = details
+  }
+}
+
+export async function openPaymentsServerErrorMiddleware(
+  ctx: AppContext,
+  next: () => Promise<unknown>
+) {
+  try {
+    await next()
+  } catch (err) {
+    const logger = await ctx.container.use('logger')
+
+    const baseLog = {
+      method: ctx.req.method,
+      path: ctx.path
+    }
+
+    if (err instanceof OpenPaymentsServerRouteError) {
+      logger.info(
```

Comment:
info because its "expected" behavior

Reply:
just digesting this out loud... looks like the paths are:

error is `OpenPaymentsServerRouteError`:
- logs at info level with details
- throws with status and message

error is `OpenAPIValidatorMiddlewareError`:
- logs at info level with details
- throws with status and message

anything else:
- logs at error level with details
- throws generic 500